apps/server/src/modules/common-cartridge/utils.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { InternalServerErrorException } from '@nestjs/common';
+import { ObjectID } from 'bson';
+import { CommonCartridgeVersion } from './common-cartridge.enums';
+import {
+	buildXmlString,
+	checkDefined,
+	checkIntendedUse,
+	createElementTypeNotSupportedError,
+	createIdentifier,
+	createResourceTypeNotSupportedError,
+	createVersionNotSupportedError,
+} from './utils';
+
+describe('CommonCartridgeUtils', () => {
+	describe('buildXmlString', () => {
+		it('should create xml string', () => {
+			const xml = buildXmlString({ root: { child: 'value' } });
+
+			expect(xml).toBe('<?xml version="1.0" encoding="UTF-8"?>\n<root>\n    <child>value</child>\n</root>');
+		});
+	});
+
+	describe('createVersionNotSupportedError', () => {
+		describe('when creating error', () => {
+			it('should return error with message', () => {
+				const error = createVersionNotSupportedError(CommonCartridgeVersion.V_1_0_0);
+
+				expect(error).toBeInstanceOf(InternalServerErrorException);
+				expect(error.message).toBe('Common Cartridge version 1.0.0 is not supported');
+			});
+		});
+	});
+
+	// AI next 16 lines
+	describe('checkDefined', () => {
+		describe('when checking value', () => {
+			it('should return value when value is defined', () => {
+				const value = 'value';
+
+				expect(checkDefined(value, 'value')).toBe(value);
+			});
+
+			it('should throw error when value is undefined', () => {
+				expect(() => checkDefined(undefined, 'value')).toThrow(new Error('value is null or undefined'));
+			});
+
+			it('should throw error when value is null', () => {
+				expect(() => checkDefined(null, 'value')).toThrow(new Error('value is null or undefined'));
+			});
+		});
+	});
+
+	describe('checkIntendedUse', () => {
+		describe('when checking supported intended use', () => {
+			it('should not throw', () => {
+				const intendedUse = 'unspecified';
+
+				expect(() => checkIntendedUse(intendedUse, ['unspecified'])).not.toThrow();
+			});
+
+			it('should throw error when intended use is not supported', () => {
+				const intendedUse = 'unsupported';
```

Comment:
This test check what is happen when a empty array is passed. But the explizit test what is happen when a different string is passed is missed. Yes it is implicit tested. But for the utils it should documentated explicit over the test how it is work. 
If i implement in code a empty array should always throw a error than the different string is not tested.

Reply:
Done.